src/Microsoft.NET.Sdk.Razor/Microsoft.NET.Sdk.Razor.csproj

```diff
@@ -41,4 +34,64 @@
       <Link>Shared\Client.cs</Link>
     </Compile>
   </ItemGroup>
+
+  <Target Name="LayoutDependencies" BeforeTargets="Build" Condition="'$(IsInnerBuild)' != 'true'">
+    <!-- Layout tasks, compiler, and extensions in the sdk-output folder. The entire folder structure gets packaged as-is into the SDK -->
+    <MSBuild
+      Projects="..\Microsoft.AspNetCore.Razor.Tools\Microsoft.AspNetCore.Razor.Tools.csproj"
+      Properties="PublishDir=$(MSBuildProjectDirectory)\$(SdkOutputPath)tools\netcoreapp2.0;TargetFramework=netcoreapp2.0"
+      Targets="Publish" />
+
+    <ItemGroup>
+      <MvcRazorExtenionOutput Include="..\Microsoft.AspNetCore.Mvc.Razor.Extensions\bin\$(Configuration)\netstandard2.0\Microsoft.AspNetCore.Mvc.Razor.Extensions.dll" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(MvcRazorExtenionOutput)" DestinationFolder="$(SdkOutputPath)extensions\mvc-2-1\" />
+
+    <ItemGroup>
+      <ProjectOutput Include="..\Microsoft.NET.Sdk.Razor\bin\$(Configuration)\net46*\Microsoft.NET.Sdk.Razor.Tasks.*" />
+      <ProjectOutput Include="..\Microsoft.NET.Sdk.Razor\bin\$(Configuration)\netstandard2.0*\Microsoft.NET.Sdk.Razor.Tasks.*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(ProjectOutput)" DestinationFiles="$(SdkOutputPath)tasks\%(RecursiveDir)%(FileName)%(Extension)" />
+  </Target>
+
+  <ItemGroup>
+    <SignedPackageFile Include="$(SdkOutputPath)tasks\net46\$(TargetName).dll" Certificate="$(AssemblySigningCertName)"  />
+    <SignedPackageFile Include="$(SdkOutputPath)tasks\netstandard2.0\$(TargetName).dll" Certificate="$(AssemblySigningCertName)"  />
+    <SignedPackageFile Include="$(SdkOutputPath)extensions\**\*.dll" Certificate="$(AssemblySigningCertName)"  />
+
+    <SignedPackageFile Include="$(SdkOutputPath)tools\netcoreapp2.0\rzc.dll" Certificate="$(AssemblySigningCertName)" />
+    <SignedPackageFile Include="$(SdkOutputPath)tools\netcoreapp2.0\Newtonsoft.Json.dll" Certificate="3PartyDual" />
+
+    <!-- Binaries that should be signed by corefx/roslyn -->
+    <ExcludePackageFileFromSigning Include="$(SdkOutputPath)tools\netcoreapp2.0\Microsoft.CodeAnalysis.CSharp.dll" />
+    <ExcludePackageFileFromSigning Include="$(SdkOutputPath)tools\netcoreapp2.0\Microsoft.CodeAnalysis.dll" />
+    <ExcludePackageFileFromSigning Include="$(SdkOutputPath)tools\netcoreapp2.0\runtimes\unix\lib\netstandard1.3\System.Text.Encoding.CodePages.dll" />
```

Comment:
This seems really fragile

Reply:
I agree. Maybe addressing https://github.com/aspnet/Razor/issues/2675 would fix this issue. That said, it's been copied over from the Razor.Design project so this isn't new:

https://github.com/aspnet/Razor/blob/master/src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj#L39-L42